src/components/structures/CallEventGrouper.ts

```diff
@@ -108,24 +108,34 @@ export default class CallEventGrouper extends EventEmitter {
         return [...this.events][0].getContent().call_id;
     }
 
+    private get roomId(): string {
+        return [...this.events][0]?.getRoomId();
```

Comment:
Are we always 100% certain we have at least one element in `this.events`? Seems from my short code analysis that this case should never happen, but could we add an additional check for it, just to be sure? Same would go for `callId`. Otherwise LGTM.

Reply:
We should always have at least one event, but I've added some `?` and made the return type into `string | undefined`, hopefully it's better this way?

Reply:
We should also probably have some null-checks though those should be in the `CallHandler`. I am going to do some more refactoring in that area, so this is one of the things I will look into